### PR TITLE
:bug: [open-formulieren/open-forms#4918] Pass initial_data_reference when redirecting to formstart

### PR DESCRIPTION
**Pull request by stevenbal**
Closes open-formulieren/open-forms#4918

### CODE REVIEW

File: src/components/Form.spec.jsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
-import {render, screen, waitFor, waitForElementToBeRemoved} from '@testing-library/react';
+import {act, render, screen, waitFor, waitForElementToBeRemoved} from '@testing-library/react';
```
Comment (sergei-maertens) [inline src/components/Form.spec.jsx:1]: left-over unused `act` import

File: src/components/Form.spec.jsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 import userEvent from '@testing-library/user-event';
 import messagesEN from 'i18n/compiled/en.json';
 import {IntlProvider} from 'react-intl';
-import {RouterProvider, createMemoryRouter} from 'react-router-dom';
+import {RouterProvider, createMemoryRouter, useNavigate} from 'react-router-dom';
```
Comment (sergei-maertens) [inline src/components/Form.spec.jsx:5]: useNavigate doesn't appear to be used

File: src/components/Form.spec.jsx [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 
 import {ConfigContext, FormContext} from 'Context';
 import {BASE_URL, buildForm, mockAnalyticsToolConfigGet} from 'api-mocks';
 import mswServer from 'api-mocks/msw-server';
 import {mockSubmissionPost, mockSubmissionStepGet} from 'api-mocks/submissions';
 import {routes} from 'components/App';
 
+import Form from './Form';
```
Comment (sergei-maertens) [inline src/components/Form.spec.jsx:13]: `Form` doesn't appear to be used

File: src/components/Form.spec.jsx [hunk 4]
```diff
@@ -114,3 +115,38 @@ test('Start form with object reference query param', async () => {
...
+  async ({introductionPageContent, buttonText, expectedUrl}) => {
+    mswServer.use(mockAnalyticsToolConfigGet(), mockSubmissionPost(), mockSubmissionStepGet());
+
+    render(
+      <Wrapper
+        form={buildForm({
+          loginOptions: [{identifier: 'digid', label: 'DigiD', url: 'http://mock-digid.nl/login'}],
+          introductionPageContent: introductionPageContent,
+        })}
+        initialEntry={`/?initial_data_reference=foo`}
```
Comment (sergei-maertens) [inline src/components/Form.spec.jsx:143]:
```suggestion
        initialEntry="/?initial_data_reference=foo"
```

File: src/components/Form.spec.jsx [hunk 5]
```diff
@@ -114,3 +115,38 @@ test('Start form with object reference query param', async () => {
...
+      />
+    );
+
+    await waitFor(() => {
+      const loginLink = screen.getByRole('link', {name: buttonText});
+      expect(loginLink).toHaveAttribute('href', expectedUrl);
+    });
```
Comment (sergei-maertens) [inline src/components/Form.spec.jsx:150]: you can use the `findBy*` queries, which are async (and basically put the  `waitFor` inside the library
```suggestion
    const loginLink = await screen.findByRole('link', {name: buttonText});
    expect(loginLink).toHaveAttribute('href', expectedUrl);
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Some polishing :)

File: src/components/Form.spec.jsx (hunk 1 above)
Comment (stevenbal) [reply src/components/Form.spec.jsx:1]: I'm too used to having flake8 check this in python :sweat_smile:

**Pull request merged**